### PR TITLE
add cities data migration

### PR DESCRIPTION
**Pull request by arwaawan3**

### CODE REVIEW

**Comment by arwaawan3** [comment]
still working on this..

**Comment by drewroberts** [comment]
I'm excited about this. Thank you for working on it.

**Comment by arwaawan3** [comment]
@drewroberts  So it looks like the timezones table only has some timezones, but the cities data has a lot more. I'm trying to handle the null values for timezone but the phpunit test is running out of memory.

**Comment by arwaawan3** [comment]
Okay so I finally ended up splitting the city data migration into 3 files, each with about 33k records. I ran PHPUnit test with only the first migration and it took extremely long. I then added the others and pushed my changes. I think these migrations should be ignored by the tests otherwise, once we merge this branch into main, future tests will timeout because of these migrations.
Screen Shot 2021-03-22 at 6 17 05 PM

**Comment by drewroberts** [comment]
Let's make sure there are not Null values for the timezones. All the USA timezones are in the package migration, so those Cities should all have timezones present. We may need to change the PHP name for the timezones in either one of the migration so they match.

**Comment by drewroberts** [comment]
Great, I'm glad that worked with 3 files.

**Comment by arwaawan3** [comment]
Ok, in that case I will update the data migrations. And the tests don't technically pass all together.. they just keep running.. It's probably because of the 100k records in the migration.

**Comment by arwaawan3** [comment]
@drewroberts Ok I think this is finally done! But the migration tests still takes forever to complete.

**Comment by drewroberts** [comment]
That's fine. Could you just send a screenshot that they are running green?

Then I'm going to move it to the `datamigrations` folder which will not be migrated in testing environments and only when the package is installed in an application. @pdbreen created it that capability in this PR:

- https://github.com/tipoff/support/pull/109

File: src/Models/Timezone.php [hunk 1]
```diff
@@ -18,14 +18,25 @@ class Timezone extends BaseModel
...
     }
 
+    public static function getNameFromPHP(string $php_timezone): string
+    {
+        $dateTime = new \DateTime();
+	$dateTime->setTimeZone(new \DateTimeZone($php_timezone));
+	$timezone = $dateTime->format('T');
+	if ($timezone == 'HDT' || 'HST') {
+	    return 'HAST';
+	}
```
Comment (drewroberts) [inline src/Models/Timezone.php]: Should we update the data migration for timezones?

**Comment by arwaawan3** [comment]
Yes they all complete and pass as long as you run one migration at a time. Here are the results.
cities01_test_pass
cities02_test_pass
cities03_test_pass

**Review by drewroberts — APPROVED** [review]
Thank you! 🎎

**Comment by arwaawan3** [comment]
Closes #112

**Pull request merged**